components/brave_rewards/resources/rewards_panel/lib/extension_host.ts

```diff
@@ -336,10 +360,34 @@ export function createHost (): Host {
     ])
 
     updateNotifications()
+  }
 
-    requestPublisherInfo().catch(console.error)
+  async function initialize () {
+    // Expose the state manager for debugging purposes.
+    Object.assign(window, {
+      braveRewardsPanel: { stateManager }
+    })
 
-    handleStartupParameters()
+    addListeners()
+
+    // Set a maximum time to display the loading indicator. Several calls to
+    // the `braveRewards` extension API can block on network requests. If the
+    // network is unavailable or an endpoint is unresponsive, we want to display
+    // the data that we have, rather than a stalled loading indicator.
+    setTimeout(() => { stateManager.update({ loading: false }) }, 3000)
+
+    proxy.handler.startRewards()
+
+    await loadPanelData()
+
+    if (!isWebUI()) {
+      // If we are running in the Rewards extension, then ensure that publisher
+      // info has been initialized in the Rewards database.
+      requestPublisherInfoForExtension().catch(console.error)
+    }
+
+    loadPersistedState()
+    handleRewardsPanelArgs((await proxy.handler.getRewardsPanelArgs()).args)
```

Comment:
Good question. It's possible that the panel will be opened with one set of arguments, and then before it calls `getRewardsPanelArgs` another request comes in with a different set of arguments. In that case, the panel would just display the UI corresponding to the second (later) arguments. I think that's probably preferable? What do you think?

Reply:
Yeah, I think the only "undesirable" scenario would be if a captcha occurred and then another request bumped it. But at least for adaptive captcha, showing the captcha relies on a user action already so it doesn't seem very likely to occur there.

Reply:
For the a-captacha case, will the request will get "lost", in the sense that there's no way for the user to reinitiate it? If so, then the solution might be to have some sort of UI that allows the user to go back to it. Or perhaps the panel could query the browser for current a-captcha requests on startup, instead of relying on panel args.

Reply:
I think we made the adaptive captcha sticky in https://github.com/brave/brave-core/pull/13188, so it should still appear in the panel (assuming it takes precedence in the panel display).